app/code/Magento/Sales/Model/Order.php

```diff
@@ -1692,6 +1692,22 @@ public function getAllStatusHistory()
         return $history;
     }
 
+    /**
+     * Return collection of not visible on frontend order status history items.
+     *
+     * @return array
+     */
+    public function getAdminVisibleStatusHistory(): array
+    {
+        $history = [];
+        foreach ($this->getStatusHistoryCollection() as $status) {
+            if (!$status->isDeleted() && $status->getComment() && !$status->getIsVisibleOnFront()) {
```

Comment:
Looks like we should remove the following line to see all the order history in the admin:
```suggestion
            if (!$status->isDeleted() && $status->getComment()) {
```

Reply:
The issue was not to show all comments in admin. Magento already shows all comments. I hid comments that shows on frontend.
If it 's bad idea we should close the PR and the issue.